src/core/lib/promise/latch.h

```diff
@@ -184,6 +205,70 @@ class Latch<void> {
   IntraActivityWaiter waiter_;
 };
 
+// A Latch that can have its value observed by outside threads, but only waited
+// upon from inside a single activity.
+template <typename T>
+class ExternallyObservableLatch;
+
+template <>
+class ExternallyObservableLatch<void> {
```

Comment:
Please make sure we have tests for this.

Reply:
we do